src/model/Ticket.java

```diff
@@ -1,87 +1,110 @@
 package model;
 
 import enums.SectorType;
+import util.IdGenerator;
 
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Currency;
 import java.util.Locale;
+import java.util.Objects;
 
-public class Ticket {
-    private String id;
-    private String concertHall;
-    private String eventCode;
+public class Ticket implements Printable, Identifiable {
+    private int classId;
+    private final String ticketId;
+    private final String concertHall;
+    private final String eventCode;
     private LocalDateTime time;
-    private boolean isPromo;
+    private final boolean isPromo;
     private SectorType stadiumSector;
-    private double maxAllowedBackpackWeightInKg;
+    private final double maxAllowedBackpackWeightInKg;
     private final LocalDateTime creationTime = LocalDateTime.now();
-    private BigDecimal ticketPrice;
+    private final BigDecimal ticketPrice;
 
     private static final String DATA_TIME_FORMAT = "yyyy.MM.dd, HH:mm";
 
     public Ticket() {
+        this.classId = IdGenerator.generateId();
```

Comment:
this "IdGenerator.generateId()" code is duplicated in each your class. If you use Interface, you can move this code into the interface Identifiable. Think about how to do it properly

Reply:
done